include/deal.II/lac/linear_operator.h

```diff
@@ -0,0 +1,1175 @@
+ * @endcode
+ */
+template <typename Range, typename Domain>
+LinearOperator<Range, Domain>
+operator*(const LinearOperator<Range, Domain> &op,
+          typename Domain::value_type  number)
+{
+  static_assert(
+    std::is_convertible<typename Range::value_type, typename Domain::value_type>::value,
+    "Range and Domain must have implicitly convertible 'value_type's");
+
+  return number * op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Returns the transpose linear operations of @ref op.
+ */
+template <typename Range, typename Domain>
+LinearOperator<Domain, Range>
+transpose_linop(const LinearOperator<Range, Domain> &op)
+{
+  LinearOperator<Domain, Range> return_op;
+
+  return_op.reinit_range_vector = op.reinit_domain_vector;
+  return_op.reinit_domain_vector = op.reinit_range_vector;
+
+  return_op.vmult = op.Tvmult;
+  return_op.vmult_add = op.Tvmult_add;
+  return_op.Tvmult = op.vmult;
+  return_op.Tvmult_add = op.vmult_add;
+
+  return return_op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Returns a LinearOperator that is the identity of the vector space
+ * @tref Range.
+ *
+ * The function takes an <code>std::function</code> object @ref exemplar as
+ * an argument to initialize the <code>reinit_range_vector</code> and
+ * <code>reinit_domain_vector</code> objects of the LinearOperator object.
+ */
+template <typename Range>
+LinearOperator<Range, Range>
+identity_linop(const std::function<void(Range &, bool)> &exemplar)
+{
+  LinearOperator<Range, Range> return_op;
+
+  return_op.reinit_range_vector = exemplar;
+  return_op.reinit_domain_vector = exemplar;
+
+  return_op.vmult = [](Range &v, const Range &u)
+  {
+    v = u;
+  };
+
+  return_op.vmult_add = [](Range &v, const Range &u)
+  {
+    v += u;
+  };
+
+  return_op.Tvmult = [](Range &v, const Range &u)
+  {
+    v = u;
+  };
+
+  return_op.Tvmult_add = [](Range &v, const Range &u)
+  {
+    v += u;
+  };
+
+  return return_op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Returns an object representing the inverse of the LinearOperator @p op.
+ *
+ * The function takes references @p solver and @p preconditioner to an
+ * iterative solver and a preconditioner that are used in the
+ * <code>vmult</code> and <code>Tvmult</code> implementations of the
+ * LinearOperator object.
+ *
+ * The LinearOperator object that is created stores a reference to @p
+ * solver and @p preconditioner. Thus, both objects must remain a valid
+ * reference for the whole lifetime of the LinearOperator object. Internal
+ * data structures of the @p solver object will be modified upon
+ * invocation of <code>vmult</code> or <code>Tvmult</code>.
+ */
+template <typename Solver, typename Preconditioner>
+LinearOperator<typename Solver::vector_type, typename Solver::vector_type>
+inverse_linop(Solver &solver,
+              const Preconditioner &preconditioner,
+              const LinearOperator<typename Solver::vector_type, typename Solver::vector_type> &op)
```

Comment:
The important argument  is the linear operator.the other two are just helpers. I would put the linear operator first.